server/src/test/java/io/crate/types/DateTypeTest.java

```diff
@@ -0,0 +1,54 @@
+package io.crate.types;
+
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class DateTypeTest {
+
+    @Test
+    public void test() {
+        ClassCastException thrown = assertThrows(ClassCastException.class,
+            () -> DateType.INSTANCE.implicitCast("not-a-number"));
+
+        assertTrue(thrown.getMessage().equals("Can't cast 'not-a-number' to date"));
```

Comment:
You could use `io.crate.testing.Asserts.assertThrows()` instead.

Reply:
Done